### PR TITLE
Added trigger buttons and rjoy

### PR DESCRIPTION
**Pull request by orkaboy**
Moved tests to lib instead (interface should be the same for both).

Linux code not implemented or tested!

### CODE REVIEW

File: crates/joystick/src/windows.rs [hunk 1]
```diff
@@ -18,9 +18,6 @@ impl Default for Joystick {
         device.plugin().unwrap();
 
         let gamepad = XGamepad {
-            buttons: vigem_client::XButtons!(
-                UP | DOWN | LEFT | RIGHT | LTHUMB | RTHUMB | LB | RB | GUIDE | A | B | X | Y
```
Comment (orkaboy) [inline crates/joystick/src/windows.rs:22]: Note: Setting this pressed all the buttons when the joystick is initialized, which is not what we want.

**Comment by orkaboy** [comment]
clippy/format is for stuff outside of the joystick crate, so that was likely an issue introduced in a different PR.

**Comment by Eein** [comment]
> clippy/format is for stuff outside of the joystick crate, so that was likely an issue introduced in a different PR.

clippy rules are still not consistent with how they are running - theres some weird rust stuff here.

File: crates/joystick/src/windows.rs (hunk 1 above)
Comment (Eein) [reply crates/joystick/src/windows.rs:22]: yea the xbuttons are set up for you in the driver on create, evdev requires you to set up every button you want to customize your joystick on init (for racing wheels, joysticks with only a couple of buttons etc)

**Comment by Eein** [comment]
Implemented linux - assuming the right joystick test does the opposite direction :)

**Review by Eein — APPROVED** [review]

**Pull request merged**